src/main/java/com/bookstore/mapper/ShoppingCartMapper.java

```diff
@@ -0,0 +1,48 @@
+package com.bookstore.mapper;
+
+import com.bookstore.config.MapperConfig;
+import com.bookstore.dto.cart.CartItemDto;
+import com.bookstore.dto.cart.ShoppingCartDto;
+import com.bookstore.entity.CartItem;
+import com.bookstore.entity.ShoppingCart;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Mapper(config = MapperConfig.class, uses = { UserMapper.class, CartItemMapper.class })
+public abstract class ShoppingCartMapper {
```

Comment:
```suggestion
public interface ShoppingCartMapper {
```

Reply:
I can not understand how to make this into an interface but keep the ShoppingCartMapper and the CartItemMapper logic separated, initially I put it into the same class because I did not want to make the ShoppingCartMapper an abstract class, however it does work as an abstract class as well.
I can not really find any info on how to do that but keep it as an interface, the only way that I've found on the internet is to do the following, but it currently breaks the application, and I am not sure why:

public interface ShoppingCartMapper {
    
    CartItemMapper cartItemMapper(); // apparantly this is supposed to return an instance of CartItemMapper but it can not because it is an interface and not a class

    @Mapping(target = "cartItems", ignore = true)
    @Mapping(target = "userId", source = "user.id")
    ShoppingCartDto toDto(ShoppingCart shoppingCart);

    @AfterMapping
    default void setCartItemDtos(ShoppingCart shoppingCart,
                                 @MappingTarget ShoppingCartDto shoppingCartDto) {
        Set<CartItemDto> cartItemDtos = shoppingCart.getCartItems()
                .stream()
                .map(cartItemMapper()::toDto)
                .collect(Collectors.toSet());
        shoppingCartDto.setCartItems(cartItemDtos);
    }

Reply:
When you used 'uses' in Mapper annotation framework will create method similar to 'setCartItemDtos' so there is no reason to autowire anything. So you can create an interface